lego/apps/users/views/users.py

```diff
@@ -176,3 +176,35 @@ def change_grade(self, request, *args, **kwargs):
 
     def destroy(self, request, *args, **kwargs):
         return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        permission_classes=[IsAuthenticated],
+        serializer_class=PhotoConsentSerializer,
+    )
+    def update_photo_consent(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = self.get_object()
+
+        with transaction.atomic():
+            if not request.user.has_perm(EDIT, user):
+                raise PermissionDenied(detail="Cannot update other user's consent")
+
+            photo_consent = None
+            try:
+                photo_consent = user.photo_consents.get(
+                    year=serializer.validated_data["year"],
+                    semester=serializer.validated_data["semester"],
+                    domain=serializer.validated_data["domain"],
+                )
+            except Exception:
+                pass
+
+            if photo_consent is None:
+                PhotoConsent.objects.create(**serializer.validated_data)
+            else:
+                photo_consent.is_consenting = serializer.validated_data["is_consenting"]
+                photo_consent.save()
```

Comment:
This seems like you want [`update_or_create()`](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#update-or-create)

Reply:
Yup! Thanks!